app.js

```diff
@@ -1,95 +1,144 @@
+          enabled: true,
+        },
+      },
+    });
+  }
+
+  return bucket;
 }
 
-const DELETE_BUCKET = async (action, settings)=>{
-    const projectId = parsers.string(action.params.PROJECT) || parsers.string(settings.PROJECT)
-    const credentials = parsers.json(action.params.CREDENTIALS) || parsers.json(settings.CREDENTIALS)
+async function deleteBucket(params) {
+  const {
+    projectId,
+    credentials: rawCredentials,
+    bucketName,
+  } = params;
 
-    const storageService = GoogleCloudStorage.from({
-        projectId : projectId,
-        credentials: credentials
-    })
+  const credentials = parseCredentials(rawCredentials);
+  const storageClient = new GoogleCloudStorageClient({ projectId, credentials });
 
-    const bucketname = parsers.autocomplete(action.params.NAME)
-    return storageService.deleteBucket({ bucketname: bucketname })
+  return storageClient.bucket(bucketName).delete();
 }
 
-const UPLOAD_FILE = async(action, settings)=>{
-    const projectId = parsers.string(action.params.PROJECT) || parsers.string(settings.PROJECT)
-    const credentials = parsers.json(action.params.CREDENTIALS) || parsers.json(settings.CREDENTIALS)
+async function upload(params) {
+  const {
+    projectId,
+    credentials: rawCredentials,
+    bucketName,
+    sourcePath,
+    destinationPath,
+  } = params;
+
+  const credentials = parseCredentials(rawCredentials);
+
+  await assertPathExistence(sourcePath);
+  const pathStat = await stat(sourcePath);
+
+  if (!pathStat.isFile() && !pathStat.isDirectory()) {
+    throw new Error("Unsupported path type! Path must point to a file or a directory.");
+  }
+
+  const storageClient = new GoogleCloudStorageClient({ projectId, credentials });
+
+  if (pathStat.isDirectory()) {
+    const directoryFiles = await listDirectoryRecursively(sourcePath);
+    const baseName = destinationPath ?? basename(sourcePath);
+    const filesToUpload = directoryFiles.map((fileName) => ({
+      source: join(sourcePath, fileName),
+      destination: join(baseName, fileName),
+    }));
+
+    return Promise.all(filesToUpload.map(({ source, destination }) => (
+      storageClient
+        .bucket(bucketName)
+        .upload(source, { destination })
+    )));
+  }
+
+  return storageClient
+    .bucket(bucketName)
+    .upload(sourcePath, {
+      destination: destinationPath,
+    });
+}
 
-    const storageService = GoogleCloudStorage.from({
-        projectId : projectId,
-        credentials: credentials
-    })
+function deleteFile(params) {
+  const {
+    projectId,
+    credentials: rawCredentials,
+    bucketName,
+    fileName,
+  } = params;
 
-    const bucketname = parsers.autocomplete(action.params.NAME)
-    const filePath = parsers.string(action.params.FILE_PATH)
-    return storageService.uploadFile({ bucketname: bucketname, filePath: filePath })
-}
+  const credentials = parseCredentials(rawCredentials);
+  const storageClient = new GoogleCloudStorageClient({ projectId, credentials });
```

Comment:
this part of code is repeated few times, so it could be extracted to some separate function like:
createStorageClient(rawCredentials) and parsing could be done inside, but it is a minor.